papermill/translators.py

```diff
@@ -114,8 +120,36 @@ def codify(cls, parameters, comment='Parameters'):
             content += '{}\n'.format(cls.assign(name, cls.translate(val)))
         return content
 
+    @classmethod
+    def inspect(cls, parameters_cell):
+        """Inspect the parameters cell to get a Parameter list
+
+        It must return an empty list if no parameters are found and
+        it should ignore inspection errors.
+
+        .. note::
+            ``inferred_type_name`` should be "None" if unknown (set it
+            to "NoneType" for null value)
+
+        Parameters
+        ----------
+        parameters_cell : NotebookNode
+            Cell tagged _parameters_
+
+        Returns
+        -------
+        List[Parameter]
+            A list of all parameters
+        """
+        raise NotImplementedError('parameters introspection not implemented for {}'.format(cls))
+
 
 class PythonTranslator(Translator):
+    # Pattern to capture parameters within cell input
+    PARAMETER_PATTERN = re.compile(
+        r"^(?P<target>\w[\w_]*)\s*(:\s*[\"']?(?P<annotation>\w[\w_\[\],\s]*)[\"']?\s*)?=\s*(?P<value>.*?)(\s*#\s*(type:\s*(?P<type_comment>[^\s]*)\s*)?(?P<help>.*))?$"  # noqa
```

Comment:
Really nice actually -- it even extracts the comment on the end of `foo = bar  # a comment`!

Reply:
The other way to do this for Python is to use the `ast` module (e.g. https://www.mattlayman.com/blog/2018/decipher-python-ast/) and extracting the assignments programmatically. However this won't work cross-language and could even hiccup with ipython specific capabilities with python kernels (e.g. `!run-something`). So I think regex is the KISS way to go.

Reply:
That is exactly my note in the PR description :wink: 

Note: I use a regex instead of ast for Python for two reasons: ast removes all comments but I use them to get a help comment. ast parsing for non-python language won't be available. So it seems better to use regex for all languages.

Reply:
Yep just re-enforcing the idea